### DISCUSSION TITLE
not fully transcribing

### DISCUSSION THREAD

**Discussion by hansolGib** [openai/whisper#1029] [Q&A]
hello, i'm trying to access lower-level to transcribe more than 30 sec audio file.
Since pad_or_trim only cut it down to 30 second, i decided to split the audio array into sublist, and then give it to pad_or_trim using for loop so that i can transcribe the whole audio. 
It's kinda successful, but i found, between the results of sublists, there are some speech which wasn't transcribed. 
How can i fully transcribe the audio longer than 30sec in lower level?

below is my code
```
model = whisper.load_model("base")
audios = whisper.load_audio(file)
 audios = [audios[x:x+480000] for x in range(0, len(audios), 480000)]
for temp in audios:
    audio = whisper.pad_or_trim(temp)
   mel = whisper.log_mel_spectrogram(audio).to(model.device)
   _, probs = model.detect_language(mel)
  options = whisper.DecodingOptions(fp16 = False)
  result = whisper.decode(model, mel, options)
  print(result.text)
```

**Comment by jstoone** [comment]
You shoud be able to just let the library do the trimming and padding by writing:
```python
import whisper

file = "audio.mp3"

model = whisper.load_model("base")
result = model.transcribe(file, fp16 = False})
print(result["text"])
```
> **NB:** _I have not tested the above code, so I'm sorry if there's a syntax error in the decode_options part of the code._

**Comment by namratas798** [comment]
heyy!! @hansolGib I am facing the same problem that whisper is not fully transcribing, did you get the solution?? please do tell me as well.
